woocommerce/woocommerce
# Fix: order refunds REST API endpoint reading from posts table even with HPOS active

### Changes proposed in this Pull Request:

Move the `get_objects` method, which retrieves order objects from the proper source depending on whether HPOS is active or not, from the `WC_REST_Orders_Controller` class to the `WC_REST_Orders_V2_Controller` class. This has two effects:

1. Make the `/orders` endpoint compatible with REST API v2 when HPOS is active (fixing v1 would require additional work).

2. Fix the `/orders/<id>/refunds` endpoint for v2 and v3.

Without this fix what happens is that the v2 endpoint for orders and the v2 and v3 endpoints for refunds always retrieve data from the posts table even when HPOS active (since the controller classes are ultimately based on `WC_REST_Posts_Controller` and any non-overridden behavior relies in the posts table).

Moving the method in just the orders class fixes the refunds too because the class hierarchy is `WC_REST_Order_Refunds_Controller extends WC_REST_Order_Refunds_V2_Controller extends WC_REST_Orders_V2_Controller` (and `WC_REST_Orders_Controller extends WC_REST_Orders_V2_Controller`).

Closes #35855.


### How to test the changes in this Pull Request:

1. Enable HPOS and make the orders table authoritative, but **disable the table synchronization**.

![image](https://user-images.githubusercontent.com/937723/210763240-e4efbcea-5a72-41ba-b3d5-4d9aad126835.png)

2. Create a completed new order with a product, then create a refund for the order.

3. Verify that the REST API endpoint `/wp-json/wc/v3/orders` includes the order you just created, and the same for the `v2` endpoint (without the fix that's true for the `v3` endpoint only).

4. Verify that the REST API endpoint `/wp-json/wc/v3/orders/<order id>/refunds` is returning the refund you created for the order, and the same for the `v2` endpoint (without the fix an empty array is returned for both endpoints).

### Other information:




### FOR PR REVIEWER ONLY:
